File: arrays/baseballGame.py
# https://leetcode.com/problems/baseball-game/

class Solution:
    
    # Runtime: 52 ms, faster than 63.73% of Python3 online submissions for Baseball Game.
    # Memory Usage: 14.2 MB, less than 34.49% of Python3 online submissions for Baseball Game.
    
    def calPoints(self, ops):
        
        
        result = []
        
        for i in ops:
            
            if i == "C":
                result.pop()
            elif i == "D":
                result.append(result[-1]*2)
            elif i == "+":
                result.append(result[-2]+result[-1])
            else:
                result.append(int(i))
                
        return sum (result)

File: arrays/baseballGame.js
// https://leetcode.com/problems/baseball-game/
// Runtime: 94 ms, faster than 58.80% of JavaScript online submissions for Baseball Game.
// Memory Usage: 47.6 MB, less than 7.14% of JavaScript online submissions for Baseball Game.
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function (operations) {
  let solution = [];
  operations.forEach((val) => {
    if (!!Number(val)) {
      solution.push(Number(val));
    } else if (val === "C") {
      solution.pop();
    } else if (val === "D") {
      solution = [...solution, solution[solution.length - 1] * 2];
    } else if (val === "+") {
      solution = [
        ...solution,
        solution[solution.length - 1] + solution[solution.length - 2],
      ];
    }
  });
  return solution.reduce((acc, cur) => acc + cur, 0);
};
module.exports = calPoints;
